src/app/modules/main/wallet_section/filter.nim

```diff
@@ -0,0 +1,44 @@
+import strformat, sequtils, sugar
+
+import ./controller
+
+
+type Filter* = ref object
+  controller: Controller
+  addresses*: seq[string]
+  chainIds*: seq[int]
+
+proc initFilter*(
+  controller: Controller,
+): Filter =
+  result = Filter()
+  result.controller = controller
+  result.addresses = @[]
+  result.chainIds = @[]
+
+
+proc `$`*(self: Filter): string =
+  result = fmt"""WalletFilter(
+    addresses: {self.addresses},
+    chainIds: {self.chainIds},
+    )"""
+
+
+proc load*(self: Filter) =
+  let accounts = self.controller.getWalletAccounts()
+  self.addresses = @[accounts[0].address]
+  self.chainIds = self.controller.getNetworks().filter(n => n.enabled).map(n => n.chainId)
+
+proc setAddress*(self: Filter, address: string) = 
+  self.addresses = @[address]
+
+proc removeAddress*(self: Filter, address: string) =
+  if len(self.addresses) == 1 and self.addresses[0] == address:
+    let accounts = self.controller.getWalletAccounts()
+    self.addresses = @[accounts[0].address]
+    return
+  
+  self.addresses.delete(self.addresses.find(address))
```

Comment:
@alaibe ok, in that case if we plan to use `addresses` as a list maybe we could do this instead:
```
  let ind = self.addresses.find(address)
  if ind > 0 and ind < self.addresses.len:
    self.addresses.delete(ind)
    return
  error "trying to remove an address which is not in the filter"
```

Reply:
It is not an error to remove an address not in the filter, as it could be you are removing an account that is not selected?

Reply:
Yes, you're right, as it's called upon many signals emitted based on user's activity.

Reply:
Do you think I should name this function something else to be more explicit?

Reply:
Good question... actually without its first part `removeAddress` is a perfect name cause it's actually what it does, but having in mind that its first part sets the wallet default address if the last address is removed then maybe we should think about other name (cause it remove and sets, so `updateFilterAddresses` or `resolveFilterAddresses` or however)?

Reply:
But thinking now... are we sure that in all cases the default wallet account will be always at the index `0`? I don't see any code which guarantees that? That can be very very tricky at some point later.

Reply:
By default the db returns first entry first. So as soon as we don't change the db engine or don't change the order in the query then it will be that case.

Also as soon as it is not the case we will see it immediately as the status account won't be first